spec/support/controller_macros.rb

```diff
@@ -0,0 +1,16 @@
+module ControllerMacros
+  def login_admin
+    before(:each) do
+      @request.env["devise.mapping"] = Devise.mappings[:admin]
+      sign_in FactoryGirl.create(:admin)
+    end
+  end
+
+  def login_user
+    before(:each) do
+      @request.env["devise.mapping"] = Devise.mappings[:user]
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.